### PR TITLE
A few fixes for lifetime parameters

### PR DESCRIPTION
**Pull request by arielb1**
r? @nikomatsakis

### CODE REVIEW

**Comment by nikomatsakis** [comment]
LGTM, thanks @arielb1. r+ modulo the four nits.

**Comment by arielb1** [comment]
<rebased, nits fixed>

File: src/librustc/middle/free_region.rs [hunk 1]
```diff
@@ -116,10 +125,17 @@ impl FreeRegionMap {
...
+                (ty::ReStatic, ty::ReFree(ref sup_fr)) => self.is_static(sup_fr),
+
                 _ =>
                     false,
             }
         }
     }
-}
 
+    /// Determinates whether this free-region is required to be 'static
```
Comment (apasel422) [inline src/librustc/middle/free_region.rs]: s/Determinates/Determines/

**Pull request merged**